### PR TITLE
DOC: clarify loadtxt input cols requirement

### PR DESCRIPTION
**Pull request by pranabdas**
As discussed in https://github.com/numpy/numpy/pull/21854#issuecomment-1167756869. 

@seberg instead of removing the sentence, I tried to provide more clarification.

### CODE REVIEW

**Comment by seberg** [comment]
Thanks!  It would be nice to have a paragraph of text there.  The reason why I suggested to remove it was that it seems like too much detail for the first thing you read about `np.loadtxt`.

Maybe someone from the docs team has a good idea.  I don't mind the information here, but would prefer leading with more generic information (for example that loadtxt is useful to parse a CSV file given the correct parameters/delimiter).

**Comment by WarrenWeckesser** [comment]
How about moving that paragraph to the Notes section?

**Comment by rossbar** [comment]
I agree with @WarrenWeckesser 's suggestion - the Notes section seems like the correct place for this amount of detail. In principle an example could be added, but it feels strange to *recommend* this practice to users.

File: numpy/lib/npyio.py [hunk 1]
```diff
@@ -1167,9 +1165,13 @@ def loadtxt(fname, dtype=float, comments='#', delimiter=None,
...
-    This function aims to be a fast reader for simply formatted files.  The
-    `genfromtxt` function provides more sophisticated handling of, e.g.,
-    lines with missing values.
+    * This function aims to be a fast reader for simply formatted files.  The
+      `genfromtxt` function provides more sophisticated handling of, e.g.,
+      lines with missing values.
+    * Each row in the input text file must have the same number of values to be
+      able to read all values. If all rows do not have same number of values, a
+      subset of up to n-columns (where n is the least number of values present
+      in all rows) can be read by specifying the columns via `usecols`.
```
Comment (WarrenWeckesser) [inline numpy/lib/npyio.py:1174]: Nitpick: these two sections don't need to be bulleted.  Regular paragraphs with a blank line between them should be fine.  Also, change `n-columns` to `n columns`.
```suggestion
    This function aims to be a fast reader for simply formatted files.  The
    `genfromtxt` function provides more sophisticated handling of, e.g.,
    lines with missing values.

    Each row in the input text file must have the same number of values to be
    able to read all values. If all rows do not have same number of values, a
    subset of up to n columns (where n is the least number of values present
    in all rows) can be read by specifying the columns via `usecols`.
```

**Review by WarrenWeckesser — CHANGES_REQUESTED** [review]
Thank @pranabdas, looks good.  I requested a slight tweak for the formatting of the text.

The addition of the example is great.

**Comment by WarrenWeckesser** [comment]
My suggested changes were trivial, so I pushed them myself and merged.  Thanks @pranabdas!

**Pull request merged**